src/shared/logger/logger.ts

```diff
@@ -18,9 +18,38 @@ export interface Logger {
     warn(error: Error): void
     error(message: string, ...meta: any[]): void
     error(error: Error): void
+    /** Returns true if the given log level is being logged.  */
+    logLevelEnabled(logLevel: LogLevel): boolean
 }
 
-export type LogLevel = keyof Logger
+/**
+ * Log levels ordered for comparison.
+ *
+ * See https://github.com/winstonjs/winston#logging-levels :
+ * > RFC5424: severity of levels is numerically ascending from most important
+ * > to least important.
+ */
+const logLevels = new Map<LogLevel, number>([
+    ['error', 1],
+    ['warn', 2],
+    ['info', 3],
+    ['verbose', 4],
+    ['debug', 5],
+])
```

Comment:
Would probably make sense to define `LogLevel` as an enum instead, but that is a something disruptive change...